optuna/integration/allennlp.py

```diff
@@ -163,16 +360,44 @@ class AllenNLPPruningCallback(EpochCallback):
         monitor:
             An evaluation metric for pruning, e.g. ``validation_loss`` or
             ``validation_accuracy``.
+
     """
 
-    def __init__(self, trial: optuna.trial.Trial, monitor: str):
+    def __init__(
+        self,
+        trial: Optional[optuna.trial.Trial] = None,
+        monitor: Optional[str] = None,
+    ):
         _imports.check()
 
         if allennlp.__version__ < "1.0.0":
             raise Exception("AllenNLPPruningCallback requires `allennlp`>=1.0.0.")
 
-        self._trial = trial
-        self._monitor = monitor
+        if trial is not None and monitor is not None:
+            self._trial = trial
+            self._monitor = monitor
+        else:
```

Comment:
I guess we can end up here when user don't pass `trial` or `monitor`. The current exception about storage types could be misleading. How about handling this case with a different exception such as `ValueError` ?

Reply:
I'm not sure if I understand your intention correctly, I added some comments in 603308d and 2987f7d.
I divided the error into the one for storage (https://github.com/optuna/optuna/pull/1772/files#diff-194650369fb014d2ea8f428774c04276R420-R425) and the other (https://github.com/optuna/optuna/pull/1772/files#diff-194650369fb014d2ea8f428774c04276R405-R413).

Please correct me know if I misunderstand your comment.